src/chainparams.cpp

```diff
@@ -134,6 +134,10 @@ class CCustomParams : public CChainParams {
         parentGenesisBlockHash = uint256S(GetArg("-parentgenesisblockhash", "0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206"));
         initialFreeCoins = GetArg("-initialfreecoins", 0);
         initial_reissuance_tokens = GetArg("-initialreissuancetokens", 0);
+        consensus.has_parent_chain = GetBoolArg("-con_has_parent_chain", true);
+        // Either it has a parent chain or not
+        const bool parent_genesis_is_null = parentGenesisBlockHash == uint256();
+        assert(consensus.has_parent_chain + parent_genesis_is_null == 1);
```

Comment:
I would probably say "^" instead of "+" here to make it clear what you mean (either one or the other must be true, but not both), assuming they are both bools.

Reply:
Oh, apparently a popular idiom for 'logical xor' is to use !=, which would also work if they're both bools.